components/automate-cli/cmd/chef-automate/start_linx.go

```diff
@@ -1,67 +1,203 @@
-		)
-		startSupCmd.Stdout = out
-		startSupCmd.Stderr = out
-		startSupCmd.SysProcAttr = &syscall.SysProcAttr{
-			Setpgid: true,
+		if startCmdFlags.opensearch {
+			Ips := infra.Outputs.OpensearchPrivateIps.Value
+			err = checkNodes(args, sshUtil, Ips, "opensearch", writer)
 		}
-		if err := startSupCmd.Start(); err != nil {
-			return status.Annotate(err, status.HabCommandError)
+		if startCmdFlags.postgresql {
+			Ips := infra.Outputs.PostgresqlPrivateIps.Value
+			err = checkNodes(args, sshUtil, Ips, "postgresql", writer)
 		}
-	} else {
-		systemctlCmd := exec.Command("systemctl", "start", "chef-automate.service")
-		systemctlCmd.Stdout = os.Stdout
-		systemctlCmd.Stderr = os.Stderr
-		if err := systemctlCmd.Run(); err != nil {
-			return status.Annotate(err, status.ServiceStartError)
+	}
+	return nil
+}
+
+func startFrontEndNodes(args []string, sshUtil SSHUtil, Ips []string, remoteService string, cliWriter *cli.Writer) error {
+	resultChan := make(chan ResultConfigSet, len(Ips))
+	originalSSHConfig := sshUtil.getSSHConfig()
+
+	scriptCommands := "sudo chef-automate start"
+
+	for _, hostIP := range Ips {
+		newSSHConfig := &SSHConfig{
+			sshUser:    originalSSHConfig.sshUser,
+			sshPort:    originalSSHConfig.sshPort,
+			sshKeyFile: originalSSHConfig.sshKeyFile,
+			hostIP:     hostIP,
 		}
+		newSSHUtil := NewSSHUtil(newSSHConfig)
+
+		go func(args []string, remoteService string, scriptCommands string, newSSHUtil SSHUtil, resultChan chan ResultConfigSet) {
+			rc := ResultConfigSet{newSSHUtil.getSSHConfig().hostIP, "", nil}
+			output, err := newSSHUtil.connectAndExecuteCommandOnRemote(scriptCommands, true)
+			if err != nil {
+				rc.Error = err
+				resultChan <- rc
+				return
+			}
+
+			err = checkOutputForError(output)
+			if err != nil {
+				rc.Error = err
+				resultChan <- rc
+				return
+			}
+
+			rc.Output = output
+			resultChan <- rc
+		}(args, remoteService, scriptCommands, newSSHUtil, resultChan)
 	}
 
+	for i := 0; i < len(Ips); i++ {
+		result := <-resultChan
+
+		if i == 0 {
+			writer.StopSpinner()
+		}
+		printStartConnectionMessage(remoteService, result.HostIP, cliWriter)
+		if result.Error != nil {
+			printStartErrorMessage(remoteService, result.HostIP, cliWriter, result.Error)
+		} else {
+			cliWriter.Printf("Output for Host IP %s : %s", result.HostIP, result.Output+"\n")
+			printStartSuccessMessage(remoteService, result.HostIP, cliWriter)
+		}
+
+		if i < len(Ips)-1 {
+			writer.StartSpinner()
+		}
+	}
+
+	close(resultChan)
 	return nil
 }
 
-func init() {
-	RootCmd.AddCommand(startCommand)
+func checkNodes(args []string, sshUtil SSHUtil, Ips []string, remoteService string, cliWriter *cli.Writer) error {
+	if len(Ips) == 0 {
+		writer.Errorf("No %s IPs are found", remoteService)
+		return status.Errorf(1, "No %s IPs are found", remoteService)
+	}
+	var err error
+	if isManagedServicesOn() {
+		return status.Errorf(status.InvalidCommandArgsError, ERROR_SELF_MANAGED_START, remoteService)
```

Comment:
Are we not supporting the start command with Managed Services?

Reply:
@Dmaddu 
In case of Managed Services being used for PG and OS, we should not control start and stop of the service.

Reply:
Ya @kalroy, The error should be provided only for PG and OS in case of managed services, but here we are returning errors for A2 and CS as well.